client/src/components/Gallery.jsx

```diff
@@ -1,41 +1,45 @@
 import React from 'react';
 import { connect } from 'react-redux';
 
+
 function Gallery({
-  selectedStyle, displayIndex, incrementDisplayIndex, decrementDisplayIndex
+  selectedStyle, displayIndex, setDisplayIndex, incrementDisplayIndex, decrementDisplayIndex
 }) {
   let displaySlides = [];
   let displayThumbs = [];
   const pictureData = selectedStyle.photos;
   
-  function makeSlide(image) {
-    return (<img className='gallerySlide' src={image.url} alt={`${selectedStyle.name} style`} />);
-  }
-  
-  function makeThumbnailBar(image) {
-    return ((<li><img className='galleryThumbnail' src={image.thumbnail_url} alt={`${selectedStyle.name} style`} /></li>));
+  function scrollThumbnails() {
+    const thumbElement = document.querySelector('.galleryThumbView li');
+    thumbElement.style.marginLeft = thumbElement.scrollWidth * displayIndex * -1 + 'px';
   }
   
   function handleLeftArrowClick() {
     if (displayIndex !== 0) {
       decrementDisplayIndex(displayIndex--);
-    }    
+    }
+    scrollThumbnails();
   }
 
   function handleRightArrowClick() {
     if (displayIndex !== selectedStyle.photos.length - 1) {
       incrementDisplayIndex(displayIndex++);
     }
+    scrollThumbnails();
+  }
+  function handleThumbnailClick(event) {
+    setDisplayIndex(event.target.dataset.index)
   }
-  
   if (pictureData) {
-    displaySlides = pictureData.map(makeSlide);
-    displayThumbs = pictureData.map(makeThumbnailBar);
+    displaySlides = pictureData.map((image) => (<img className='gallerySlide' src={image.url} alt={`${selectedStyle.name} style`} />));
+    displayThumbs = pictureData.map((image, index) => (<li><img onClick={handleThumbnailClick} data-index={index} className='galleryThumbnail' src={image.thumbnail_url} alt={`${selectedStyle.name} style`} /></li>));
```

Comment:
although not important now. Add unique keys to any dynamically generated list. If the data does not have a unique id associated with it, perhaps you can get creative and use slice to chop off the unique part of a url to use as a key.